galley/pkg/config/analysis/analyzers/virtualservice/destinationhosts.go

```diff
@@ -112,34 +107,69 @@ func (d *DestinationHostAnalyzer) checkDestinationHost(vsNamespace string, desti
 		}
 
 		// Skip over entries not visible to the current virtual service namespace
-		if scope != util.ExportToAllNamespaces && scope != vsNamespace {
+		if scope != util.ExportToAllNamespaces && scope != sourceNs {
 			continue
 		}
 
 		seHostWithoutWildcard := strings.TrimPrefix(seHost, util.Wildcard)
 		hostWithoutWildCard := strings.TrimPrefix(host, util.Wildcard)
 
 		if strings.HasSuffix(hostWithoutWildCard, seHostWithoutWildcard) {
-			return true
+			return s
 		}
 	}
 
-	return false
+	return nil
 }
 
-func initServiceEntryHostNames(ctx analysis.Context) map[util.ScopedFqdn]bool {
-	hosts := make(map[util.ScopedFqdn]bool)
-	ctx.ForEach(metadata.IstioNetworkingV1Alpha3Serviceentries, func(r *resource.Entry) bool {
+func initServiceEntryHostMap(ctx analysis.Context) map[util.ScopedFqdn]*v1alpha3.ServiceEntry {
+	result := make(map[util.ScopedFqdn]*v1alpha3.ServiceEntry)
+
+	extractFn := func(r *resource.Entry) bool {
```

Comment:
how does this work for platform services? I'm guessing we can't assert it to `r.Item.(*v1alpha3.ServiceEntry)`

Reply:
We can, actually! Since this analysis is happening after Galley has already pre-processed the data, so regular ServiceEntry CRs and platform services both get turned into ServiceEntry proto objects. They're in different collections (ServiceEntries vs SyntheticServiceEntries) but for this purpose at least interchangeable.

Reply:
I see, thanks for letting me know.